bee-common/src/shutdown.rs

```diff
@@ -51,13 +53,13 @@ impl Shutdown {
     }
 
     /// Adds an asynchronous worker.
-    pub fn add_worker(&mut self, worker: WorkerHandle) {
-        self.workers.push(worker);
+    pub fn add_worker<T>(&mut self, worker: impl Future<Output = Result<T, WorkerError>> + Unpin + 'static) {
```

Comment:
why `T` ?

Reply:
That's the standard type name of `Result` https://doc.rust-lang.org/std/result/enum.Result.html

Reply:
yea, I know, but why not just have `impl<Output = Result<(), WorkerError>> + ....`, then we don't need `add_worker` to be generic over `T` ?

Reply:
the workers themselves produce no values I think